components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIConsumerImpl.java

```diff
@@ -2147,29 +2236,89 @@ public void updateApplication(Application application) throws APIManagementExcep
         }
     }
 
-
     /**
      * Function to remove an Application from the API Store
+     *
      * @param application - The Application Object that represents the Application
      * @throws APIManagementException
      */
     public void removeApplication(Application application) throws APIManagementException {
-        APIManagerConfiguration config = ServiceReferenceHolder.getInstance().
-                getAPIManagerConfigurationService().getAPIManagerConfiguration();
+        boolean isTenantFlowStarted = false;
+        int applicationId = application.getId();
+
+        if (tenantDomain != null && !MultitenantConstants.SUPER_TENANT_DOMAIN_NAME.equals(tenantDomain)) {
```

Comment:
Should apply malintha's previous comment[start tenant flow in try block] here as well.